src/models/instance.ts

```diff
@@ -34,6 +35,17 @@ export default class Instance {
         return new Instance(initialData);
     }
 
+    public static getOrCreate(data: Data, encryptionKey: string): Instance {
```

Comment:
`data` may be undefined: `getOrCreate(data: Data | undefined, encryptionKey: string`. In vaccination we use the type `Maybe` (usual in many languages): `type Maybe<T> = T | undefined;`. Both ways are ok.